File: run_server.js
#!/usr/bin/env node
/**
 * Cyber Securivox - Simple Node.js Server Launcher
 * Run this file directly in VS Code to start the cybersecurity platform.
 */

const http = require('http');
const fs = require('fs');
const path = require('path');
const { exec } = require('child_process');

// Configuration
const PORT = 3000;
const HOST = 'localhost';
const URL = `http://${HOST}:${PORT}`;

// MIME types for different file extensions
const mimeTypes = {
    '.html': 'text/html',
    '.css': 'text/css',
    '.js': 'text/javascript',
    '.json': 'application/json',
    '.png': 'image/png',
    '.jpg': 'image/jpeg',
    '.gif': 'image/gif',
    '.ico': 'image/x-icon',
    '.svg': 'image/svg+xml'
};

function openBrowser() {
    console.log(`🌐 Opening browser to: ${URL}`);
    
    // Cross-platform browser opening
    const command = process.platform === 'win32' ? 'start' : 
                   process.platform === 'darwin' ? 'open' : 'xdg-open';
    
    exec(`${command} ${URL}`, (error) => {
        if (error) {
            console.log('💡 Please manually open your browser to:', URL);
        }
    });
}

function serveFile(req, res) {
    let filePath = req.url === '/' ? '/index.html' : req.url;
    filePath = path.join(__dirname, filePath);
    
    const extname = path.extname(filePath);
    const contentType = mimeTypes[extname] || 'text/plain';
    
    fs.readFile(filePath, (error, content) => {
        if (error) {
            if (error.code === 'ENOENT') {
                res.writeHead(404, { 'Content-Type': 'text/html' });
                res.end('<h1>404 - File Not Found</h1>', 'utf-8');
            } else {
                res.writeHead(500);
                res.end(`Server Error: ${error.code}`, 'utf-8');
            }
        } else {
            res.writeHead(200, { 'Content-Type': contentType });
            res.end(content, 'utf-8');
        }
    });
}

function startServer() {
    console.log('='.repeat(50));
    console.log('🛡️  CYBER SECURIVOX - CYBERSECURITY PLATFORM');
    console.log('='.repeat(50));
    console.log();
    console.log(`🚀 Starting Node.js server on ${HOST}:${PORT}`);
    console.log(`📂 Serving files from: ${__dirname}`);
    console.log(`🌐 Access the platform at: ${URL}`);
    console.log();
    console.log('📋 Available Pages:');
    console.log('   • Home Page: /');
    console.log('   • Habit Tracker: /tracker.html');
    console.log('   • Link Checker: /link-checker.html');
    console.log('   • Password Tester: /password-tester.html');
    console.log('   • CyberCoach Chatbot: /chatbot.html');
    console.log('   • Learning Center: /learn.html');
    console.log('   • Security Scanner: /security-scanner.html');
    console.log();
    console.log('⚠️  To stop the server: Press Ctrl+C');
    console.log('='.repeat(50));
    console.log();
    
    const server = http.createServer(serveFile);
    
    server.listen(PORT, HOST, () => {
        console.log('✅ Server started successfully!');
        console.log(`🔗 Click here: ${URL}`);
        console.log();
        
        // Open browser after 2 seconds
        setTimeout(openBrowser, 2000);
    });
    
    server.on('error', (error) => {
        if (error.code === 'EADDRINUSE') {
            console.log(`❌ Error: Port ${PORT} is already in use!`);
            console.log('💡 Try closing other applications or change the PORT variable');
        } else {
            console.log(`❌ Server error: ${error.message}`);
        }
    });
    
    // Handle graceful shutdown
    process.on('SIGINT', () => {
        console.log('\n🛑 Server stopped by user');
        console.log('👋 Thank you for using Cyber Securivox!');
        server.close(() => {
            console.log('🔒 Server shutdown complete.');
            process.exit(0);
        });
    });
}

// Start the server
startServer();

File: run_server.py
#!/usr/bin/env python3
"""
Cyber Securivox - Simple Server Launcher
Run this file directly in VS Code to start the cybersecurity platform.
"""

import http.server
import socketserver
import webbrowser
import os
import sys
import time
from threading import Timer

# Configuration
PORT = 8000
HOST = "localhost"
URL = f"http://{HOST}:{PORT}"

def open_browser():
    """Open the default browser after a short delay"""
    print(f"🌐 Opening browser to: {URL}")
    webbrowser.open(URL)

def start_server():
    """Start the HTTP server"""
    try:
        # Change to the directory containing this script
        os.chdir(os.path.dirname(os.path.abspath(__file__)))
        
        print("=" * 50)
        print("🛡️  CYBER SECURIVOX - CYBERSECURITY PLATFORM")
        print("=" * 50)
        print()
        print(f"🚀 Starting server on {HOST}:{PORT}")
        print(f"📂 Serving files from: {os.getcwd()}")
        print(f"🌐 Access the platform at: {URL}")
        print()
        print("📋 Available Pages:")
        print("   • Home Page: /")
        print("   • Habit Tracker: /tracker.html")
        print("   • Link Checker: /link-checker.html")
        print("   • Password Tester: /password-tester.html")
        print("   • CyberCoach Chatbot: /chatbot.html")
        print("   • Learning Center: /learn.html")
        print("   • Security Scanner: /security-scanner.html")
        print()
        print("⚠️  To stop the server: Press Ctrl+C")
        print("=" * 50)
        print()
        
        # Open browser after 2 seconds
        Timer(2.0, open_browser).start()
        
        # Create and start the server
        with socketserver.TCPServer((HOST, PORT), http.server.SimpleHTTPRequestHandler) as httpd:
            print(f"✅ Server started successfully!")
            print(f"🔗 Click here: {URL}")
            print()
            httpd.serve_forever()
            
    except KeyboardInterrupt:
        print("\n")
        print("🛑 Server stopped by user")
        print("👋 Thank you for using Cyber Securivox!")
        
    except OSError as e:
        if "Address already in use" in str(e):
            print(f"❌ Error: Port {PORT} is already in use!")
            print("💡 Try closing other applications or use a different port")
        else:
            print(f"❌ Error starting server: {e}")
        
    except Exception as e:
        print(f"❌ Unexpected error: {e}")
        
    finally:
        print("\n🔒 Server shutdown complete.")
        input("Press Enter to exit...")

if __name__ == "__main__":
    start_server()
